cmd/go-vcs/go-vcs.go

```diff
@@ -309,23 +310,38 @@ func main() {
 		}
 
 	case "branches":
-		if len(args) != 0 {
-			log.Fatal("branches takes no arguments.")
+		if len(args) > 1 {
+			log.Fatal("branches takes 0 or 1 arguments.")
+		}
+
+		// Open using go/vcs to figure out VCS type (git, hg).
+		r := vcs2.New(".")
+		if r == nil {
+			log.Fatalln("no supported vcs found in cwd")
 		}
 
-		repo, err := vcs.Open("hg", ".")
+		repo, err := vcs.Open(r.Type().VcsType(), ".")
```

Comment:
I am aware of my deviation from the Go style in this instance, and it is a valid point. Thanks!

Most of my personal Go code breaks that Go style suggestion (but only that one). I had some reasons for doing it. I may want to revision my decision about it in the future.